### ISSUE TITLE
Exponent disappearing in tabular environment with table format and exponent-mode=threshold

### ISSUE THREAD

**Issue by JelleBosmansBE** [josephwright/siunitx#681]
Experiencing problems with S columns in the tabular environment when the input is in scientific notation. Both positive and negative exponents are being ignored, leading to incorrect numerical values in the table produced. 

MWE: 

```\documentclass{article}
\usepackage{siunitx}
\begin{document}
\sisetup{round-mode=places,round-precision=1}
\begin{tabular}{S[table-format=+3.1,exponent-mode=fixed,fixed-exponent=0]| S[table-format=+3.1,exponent-mode=threshold]}
	1.307050970250419e+02  & 1.307050970250419e+02  \\
	-1.554096754594953e+02 & -1.554096754594953e+02 \\
	60.909080403561550     & 60.909080403561550     \\
	6.909080403561550e-02  & 6.909080403561550e-02
\end{tabular}
\end{document}
```

This code produces

![image](https://github.com/josephwright/siunitx/assets/113995601/a4f75323-981a-429d-892e-9bbe5aa50318)

Whereas I would expect the same result from both columns.

**Comment by josephwright** [comment]
You've not allowed any space in `table-format` for the exponents: you'd need `table-format = +3.1e-1` here.

**Comment by JelleBosmansBE** [comment]
Upon further examination it appear the issue stems from the fact that I misinterpreted how the exponent-mode=threshold works. When the value is within the thresholds, it uses same exponent mode as the input, which is in this case scientific notation. However there is no room for scientific notation in the table format I defined, which is why it's dropping it.
Suggestion: If possible issue a warning for this behavior through the log, otherwise tell other users like myself to RTFM :-)
